core/client/src/main/java/alluxio/client/block/StreamFactory.java

```diff
@@ -103,13 +103,14 @@ public static BlockInStream createRemoteBlockInStream(FileSystemContext context,
    * @param blockSize the block size
    * @param blockStart the start position of the block in the UFS file
    * @param address the worker network address
+   * @param mountId the id of the mount that this file is mapped to
    * @param options the in stream options
    * @return the input stream
    */
   public static BlockInStream createUfsBlockInStream(FileSystemContext context, String ufsPath,
-      long blockId, long blockSize, long blockStart, WorkerNetAddress address,
+      long blockId, long blockSize, long blockStart, WorkerNetAddress address, long mountId,
```

Comment:
Reading your comment twice. If you are asking whether to have `mountId` embedded in `ufsPath`,  I would prefer not. I think `ufsPath` should be an identity without Alluxio information, and it can be converted from / to a readable string.